src/isar_turtlebot/robotinterface.py

```diff
@@ -39,21 +41,61 @@ def __init__(self):
         self.bridge: RosBridge = RosBridge()
 
     def schedule_step(self, step: Step) -> Tuple[bool, Optional[Any], Optional[Joints]]:
-        self._publish_task(step=step)
-        previous_run_id: str = self._get_run_id()
-        run_id: str = self._wait_for_updated_task(previous_run_id=previous_run_id)
+        run_id: str = self._publish_task(step=step)
         return True, run_id, None
 
     def _publish_task(self, step: Step) -> None:
         if isinstance(step, DriveToPose):
+            self.current_task = "navigation"
+            previous_run_id: str = self._get_run_id()
             self._publish_navigation_task(pose=step.pose)
+            run_id: str = self._wait_for_updated_task(previous_run_id=previous_run_id)
 
+            return run_id
         elif isinstance(step, TakeImage):
```

Comment:
Should the function return something if it does not return  under "if" and "elif"?